File: ShopCore/src/main/java/shop/core/enums/PetTypes.java
package shop.core.enums;

import java.math.BigDecimal;

public enum PetTypes {
	
	Dog(new BigDecimal(1000), 0),
	Cat(new BigDecimal(400), 1),
	Cow(new BigDecimal(6000), 2),
	Donkey(new BigDecimal(3000), 3);
	
	private BigDecimal price;
	private int petNumber;
	
	PetTypes(BigDecimal price, int petNumber){
		this.price = price;
		this.petNumber = petNumber;
	}
	
	public BigDecimal getPrice() {
		return price;
	}
	public void setPrice(BigDecimal price) {
		this.price = price;
	}
	public int getPetNumber() {
		return petNumber;
	}
	public void setPetNumber(int petNumber) {
		this.petNumber = petNumber;
	}
	
	public static PetTypes getTypeByNumber(int petNumber){
		for(PetTypes type: PetTypes.values()){
			if(petNumber == type.getPetNumber())
				return type;
		}
		return null;
	}
	
	public static PetTypes getTypeByName(String name){
		for(PetTypes type: PetTypes.values()){
			if(type.name().equalsIgnoreCase(name))
				return type;
		}
		return null;
	}
	
}

File: ShopClientDotNet/ShopClientDotNet/core/PetTypes.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ShopClientDotNet
{
    public class PetTypes : Attribute
    {
        public PetTypes(string name, int order, double price)
        {
            this.name = name;
            this.order = order;
            this.price = new Decimal(price);
        }
        private string name;
        private int order;
        private Decimal price;

        public string Name
        {
            get
            {
                return this.name;
            }
        }

        public int Order
        {
            get
            {
                return this.order;
            }
        }

        public Decimal Price
        {
            get
            {
                return this.price;
            }
        }
            

      
    }

    public enum Pets
    {
        [PetTypes("Dog", 0, 1000)] Dog,
        [PetTypes("Cat", 1, 400)] Cat,
        [PetTypes("Cow", 2, 6000)] Cow,
        [PetTypes("Donkey", 3, 3000)] Donkey
    
    }

    
}
